[Bugfix] Fix newContentElementWizard url in record context menu

thomasrawiel opened the pull request:
See issue #537

When a record is placed inside a container, the URL to create the New Content Element Wizard is wrong, thus showing the wrong list of available content elements (e.g. if using EXT:content_defender)
When tx_container_parent is missing in that URL, the changes in B13\Container\Listener\ModifyNewContentElementWizardItems are not applied.

This PR adds the tx_container_parent attribute to the New Content Element Wizard URL in the Record Context Menu

baschny commented:
@thomasrawiel very cool, this works like a charm!

achimfritz commented:
Hi @thomasrawiel , thanks for PR, (did not know/used that api....), i do not understand why die Class is used from TYPO3? It is either registered nor a XClass? (https://docs.typo3.org/m/typo3/reference-coreapi/11.5/en-us/ApiOverview/Backend/ContextualMenu.html#step-1-item-provider-registration?)

Some Points:
- unfortunaly the CI Pipeline fails
- i would prefer to change namespace to B13\Container\Backend\ContextMenu;
- declare strict statement ist missing

thomasrawiel commented:
@achimfritz For TYPO3 12 and up: If you have autoconfigure: true set in your extension's Services.yaml file all classes implementing [\TYPO3\CMS\Backend\ContextMenu\ItemProviders\ProviderInterface ](https://api.typo3.org/12.4/classes/TYPO3-CMS-Backend-ContextMenu-ItemProviders-ProviderInterface.html) get registered as context menu items automatically

For TYPO3 11 I have added the registration in ext_localconf.php.

I also moved the namespace and added a strict_types

For the CI pipelines I understand there is the method getLanguageField not (yet) implemented in TYPO3 11, I added a check for that

achimfritz reviewed (approved):
thanks, 👍
